proposals/xxxx-package-manager-performance-testing.md

```diff
@@ -0,0 +1,63 @@
+# Package Manager Performance Testing Conventions
+
+* Proposal: [SE-XXXX](xxx-package-manager-performance-testing.md)
+* Author: [Ankit Aggarwal](https://github.com/aciidb0mb3r)
+* Review Manager: TBD
+* Status: Discussion
+
+## Introduction
+
+This is a proposal for adding package manager support for building and running performance tests in release mode.
+
+## Motivation
+
+The package manager supports building and running tests using the `swift test` subtool. The sources are built in debug mode with swift compiler's
+`-enable-testing` flag allowing testable imports in test modules (which gives it internal level access to the imported module). 
+This works well for normal unit tests, however for performance tests we are usually interested in release build test results which takes advantages of compiler 
+optimizations.
+
+We can provide an option to compile and run the tests in release mode with `-enable-testing` flag. However that will disable some of the compiler optimizations, which 
+may be of interest to packages having very performance sensitive code (which critically depends on those optimization).
+
+Since performance tests usually take more time to run than other tests, there is also a requirement to have a way to run performance tests separate from unit tests.
+As a concrete example, package manager's own performance tests are disabled from running on Swift CI.
+
+## Proposed solution
+
+We propose to extend the test modules conventions and consider a directory as "performance test module" which has a suffix "PerformanceTests". The tests in these modules will only run when `--performance` flag is passed to `swift test`.
+
+This command will run performance tests in debug configuration:
+
+```
+$ swift test --performance
+```
+
+And this command will run performance tests in release configuration:
+
+```
+$ swift test --performance -c release
+```
+
+It may be desirable to run performance test in release mode with testing enabled. We propose to add a new boolean property (`enablePerformanceTestability`) in manifest to turn on testability for performance tests in release mode, so packages which want to use the feature in performance tests can do so without passing `-Xswiftc -enable-testing` flags. The default value of this property will be `false`:
```

Comment:
We need to specify what happens when a user asks for performance tests, but it would require multiple builds (because some modules want testability and some don't).

The behavior I personally want is that we find a way to support a mode where if a user has both white box and black box tests, we do the build twice (so we can build all the testability requiring tests) to get the maximal coverage.